c/driver/postgresql/connection.cc

```diff
@@ -170,13 +241,10 @@ AdbcStatusCode PostgresConnectionGetObjectsImpl(
 
     PqResultHelper result_helper = PqResultHelper{conn, query.buffer};
     StringBuilderReset(&query);
-    pg_result* result = result_helper.Execute();
 
-    ExecStatusType pq_status = PQresultStatus(result);
-    if (pq_status == PGRES_TUPLES_OK) {
-      int num_rows = PQntuples(result);
-      for (int row = 0; row < num_rows; row++) {
-        const char* db_name = PQgetvalue(result, row, 0);
+    if (result_helper.Status() == PGRES_TUPLES_OK) {
```

Comment:
Hmm, I suppose the original code had the same issue, but we should be returning an error if the status here is not OK.

Reply:
Ah, I see now.

That's probably not the right code (and we should have error details) + I think that should really pair with the `if (!catalog)` but we can fix that up later.